### DISCUSSION TITLE
Ask

### DISCUSSION THREAD

**Discussion by edo3186** [wazuh/wazuh#16416] [Q&A]
hello, I'm having a problem, where my agent can't be registered, does the port on the agent have to open ports 1515 and 1514?

**Comment by jctello** [comment]
Hi @edo3186 ,
The agent only needs to have outbound connectivity on ports 1515 and 1514.
It needs to be able to connect to the manager on those ports to register (port 1515) and communicate afterwards (1514). 

You may verify connectivity as explained here in our documentation: https://documentation.wazuh.com/current/user-manual/agent-enrollment/troubleshooting.html#testing-communication-with-the-wazuh-manager
- For Linux and macOS: `nc -zv <MANAGER_IP> 1514 1515 55000`
- For Windows: `(new-object Net.Sockets.TcpClient).Connect("<MANAGER_IP>", 1514)`, `(new-object Net.Sockets.TcpClient).Connect("<MANAGER_IP>", 1515)`

**Reply by edo3186** [reply]
@jctello  My port agen not succes,
1515 (tcp) failed Connection refused

1514 (tcp) failed Connection refused

**Reply by edo3186** [reply]
![image](https://user-images.githubusercontent.com/99638987/225856065-9e114a62-9d09-4f9a-ae1b-6e6b3f789f51.jpeg)

**Reply by jctello** [reply]
In this case you must then review:
- If the manager's service is running (`systemctl status wazuh-manager`)
- If there's a firewall restriction blocking these ports between the agent and manager

**Reply by edo3186** [reply]
what should i do if it is blocked by firewall

**Reply by jctello** [reply]
It will strongly depend on which firewall is blocking the request. 
For example if the culprit is the manager's `firewalld` then running:
`firewall-cmd --permanent --zone=public --add-port={1514/tcp,1515/tcp} && firewall-cmd --reload` will unblock these ports.
But if it is a network firewall you must then refer to the documentation of that device.

**Comment by edo3186** [comment]
Hello, if our IP is blocked on wazuh itself, there is a solution not to open the IP on wazuh?
